hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/WriteBehindQueue.java

```diff
@@ -20,108 +20,89 @@
 import java.util.List;
 
 /**
- * Main contract for write behind queues
- * which are used for map store operations.
+ * A specific queue implementation which is used for write-behind-store operations.
+ * Also supports some filtering methods e.g. {@link #getFrontByTime}, {@link  #getFrontByNumber}
  *
- * @param <E> Type of entry to be stored.
+ * @param <E> the type of element to be stored in this queue.
  */
 public interface WriteBehindQueue<E> {
 
     /**
-     * adds to the end.
+     * Inserts to the end of this queue.
      *
-     * @param e item to be offered
-     * @return <code>true</code> if added, <code>false</code> otherwise.
+     * @param e element to be offered
      */
-    boolean offer(E e);
+    void addEnd(E e);
 
     /**
-     * Gets item.
+     * Inserts collection of elements to the front of this queue.
      *
-     * @param e item to be offered
-     * @return corresponding item from queue or null.
+     * @param collection collection of elements to be inserted in front of this queue.
      */
-    E get(E e);
-
-    /**
-     * Gets first item in queue.
-     *
-     * @return corresponding item from queue or null.
-     */
-    E getFirst();
-
-    /**
-     * removes head of the queue.
-     */
-    void removeFirst();
-
-    int size();
-
-    void clear();
-
-    /**
-     * @return A copy of queue at that moment. Returned copy has same characteristics with the original.
-     */
-    WriteBehindQueue<E> getSnapShot();
+    void addFront(Collection<E> collection);
```

Comment:
API seems asymmetric, there's `void addEnd(E e)` method but no `void addFront(E e)`. There's `void addFront(Collection<E> collection)` but not `void addEnd(Collection<E> collection)`. 

Also I prefer `addFirst()` and `addLast()` names because that's the naming used in JDK.